pkg/resource/edit/operations.go

```diff
@@ -88,3 +92,61 @@ func LocateResource(snap *deploy.Snapshot, urn resource.URN) []*resource.State {
 
 	return resources
 }
+
+// RenameStack changes the `stackName` component of every URN in a snapshot. In addition, it rewrites the name of
+// the root Stack resource itself.
+func RenameStack(snap *deploy.Snapshot, newName tokens.QName) error {
+	rewriteUrn := func(u resource.URN) resource.URN {
+		// The pulumi:pulumi:Stack resource's name component is of the form `<project>-<stack>` so we want
+		// to rename the name portion as well.
+		if u.QualifiedType() == "pulumi:pulumi:Stack" {
```

Comment:
Do you know how common it is for resources other than `Stack` to have names that contain the stack name? If it's common, maybe it would be nice here to find-replace the stack name within those URNs as well?

Reply:
> If it's common, maybe it would be nice here to find-replace the stack name within those URNs as well?

Yeah, I don't know of other common places but we can certainly add them as they crop up.  Hopefully the `aliases` feature will also make it easy for folks to handle the fallout of a rename.